### DISCUSSION TITLE
deploy error on heroku

### DISCUSSION THREAD

**Discussion by Axitvadi** [fastify/fastify#4104] [Q&A] [answered]
**Error R10 (Boot timeout) -> Web process failed to bind to $PORT within 60 seconds of launch** 

The Heroku logs indicate that once the deployed successfully, Server running on port but, the above error occurs when I call any endpoint. 

In addition, I refer to the old open issue regarding this issue, but I am not able to resolve my issue with that implementation that's why i open this ticket 
It would be great if you could help me on this

**Comment by mcollina** [comment]
Thanks for reporting!

Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Reply by Axitvadi** [reply]
app running on localhost successfully but on heroku it shows [Error] R10 (Boot timeout) -> Web process failed to bind to $PORT within 60 seconds of launch
this is github repository link : https://github.com/smit-code/fastest-rest-countries-states-cities.git
if possible please review code 

great thanks !

**Comment by Eomm** [comment]
You need to change the listen call

```diff
-fastify.listen({ port: 3000, host: '127.0.0.1' }, (err, address) => {
+fastify.listen({ port, host: '0.0.0.0' }, (err, address) => {
```

**Comment by smit-code** [comment, marked as answer]
`fastify.listen({ port, host: '0.0.0.0' }, (err, address) => {`

It will default to 127.0.0.1, which cannot be accessed from outside.
Instead of 127.0.0.1, use 0.0.0.0.
